### ISSUE TITLE
Enable Etherscan Support for Diamond

### ISSUE THREAD

**Issue by 0x4007** [pavlovcik/uad-ui#322]
Depends on #300 to be completed.

We should add etherscan support for the [Diamond contract](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/src/dollar/Diamond.sol) using https://github.com/zdenham/diamond-etherscan

Corresponding [docs](https://github.com/zdenham/diamond-etherscan#how-to-support-etherscan-in-your-diamond)

What should be done:
1. Add a shell script which generates the dummy contract (step 1 from the [docs](https://github.com/zdenham/diamond-etherscan#how-to-support-etherscan-in-your-diamond))
2. Add dummy contract deployment to the development [deploy scripts](https://github.com/ubiquity/ubiquity-dollar/tree/development/packages/contracts/scripts/deploy/dollar/solidityScripting)
3. Add [EtherscanFacet](https://github.com/zdenham/diamond-etherscan/blob/main/contracts/facets/DiamondEtherscanFacet.sol) to our Diamond contract
4. In the dummy [deployment script](https://github.com/ubiquity/ubiquity-dollar/tree/development/packages/contracts/scripts/deploy/dollar/solidityScripting) set the proxy implementation to point to the deployed dummy

This way we would have an "etherscan friendly" development deploy which we would later modify for production deployment

**Comment by 0xautohub** [comment]
This is really useful. Awesome 👍

**Comment by zgorizzo69** [comment]
interesting I am willing to do that !

**Comment by 0xautohub** [comment]
@pavlovcik @sergfeldman I hope to implement this as a continuous work of diamond

**Comment by 0xautohub** [comment]
And if @zgorizzo69 you have your own work and don't mind..

**Comment by sergfeldman** [comment]
> @pavlovcik @sergfeldman I hope to implement this as a continuous work of diamond

@sunny0714 Thank you!! 
In some days @pavlovcik will return from the conference, we will discuss further implementation of Diamond, and after that it will be clear who will take this issue.

**Comment by 0x4007** [comment]
I think its fine if @sunny0714 tries this bounty. Good luck!

**Comment by sergfeldman** [comment]
FYI https://github.com/zdenham/diamond-etherscan
https://etherscan.diamonds.dev/

**Comment by 0x4007** [comment]
The #300 dependency is ready

**Comment by 0xautohub** [comment]
@pavlovcik Has the completed diamond already deployed on goerli testnet?

**Comment by 0x4007** [comment]
I don't believe so but either you or somebody on the team can do the deployment there. You want to give it a try and tag us if you can't figure it out?

**Comment by 0xautohub** [comment]
I have checked the codebase related to deployment a bit and found that there were many updates.
A series of Solidity deployment scripts looks good, and I got to know that DevelopmentDeploy.ts dealt with the deployment on anvil testnet while impersonating curve whale account
And had a wonder if there is any problem when deploying on goerli testnet, and if there needs a separate deployment script.
Because I was away from the Ubiquity project for several months, it's not easy to jump on the implementation right away. Hope you guys to guide me to deploy the Ubiquity core on goerli testnet

**Comment by 0x4007** [comment]
> Hope you guys to guide me to deploy the Ubiquity core on goerli testnet

@rndquu are you able to handle this?

**Comment by rndquu** [comment]
> > Hope you guys to guide me to deploy the Ubiquity core on goerli testnet
> 
> @rndquu are you able to handle this?

I guess this task is still blocked because we don't have a testnet (i.e. goerli) deployment script yet. We have a task for [dev environments](https://github.com/ubiquity/ubiquity-dollar/issues/658) and testnet deploy script will be part of that task.

So @sunny0714 pls wait for https://github.com/ubiquity/ubiquity-dollar/issues/658 (or at least for the testnet deployment script)

**Comment by 0xautohub** [comment]
This issue is no longer available?

**Comment by rndquu** [comment]
> This issue is no longer available?

pls wait for the updated issue description

**Comment by molecula451** [comment]
it's possible this task is a follow up after #682, when we deploy the diamond to a live testnet

**Comment by molecula451** [comment]
After the recent deployment, this looks like a low priority task issue we are not looking to achieve (to my extend) in the short-term and this repo to rely on achieving the task it's kinda outdated in some sense. Closing, re-opening when really needed it
